### PR TITLE
Homotopy example

### PR DESCRIPTION
**Pull request by rli596**
Added a simple example for Homotopy animation

## Links to added or changed documentation pages

https://manimce--2959.org.readthedocs.build/en/2915/reference/manim.animation.movement.Homotopy.html

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by alembcke** [comment]
The link you provided is for the main documentation page.  Here is the link for this specific PR, which shows your added example:
https://manimce--2959.org.readthedocs.build/en/2915/reference/manim.animation.movement.Homotopy.html

File: manim/animation/movement.py [hunk 1]
```diff
@@ -44,6 +44,17 @@ class Homotopy(Animation):
...
+    Examples
+    --------
+    .. manim:: HomotopyExample
+
+        class HomotopyExample(Scene):
+            def construct(self):
+                dot = Dot()
+                homotopy = lambda x,y,z,t:[x + t, y + t**2, z]
+
+                self.play(Homotopy(homotopy, dot))
```
Comment (sparshg) [inline manim/animation/movement.py:57]: Currently it look just like shifting the dot. How about something like this waving square:
```suggestion
                square = Square()
                homotopy = lambda x, y, z, t: (x, y + 0.2 * np.sin(x + 10 * t), z)
                self.play(Homotopy(homotopy, square))
```

https://user-images.githubusercontent.com/43041139/192983555-1a602921-51fd-4b66-97b9-9ede4757fcda.mp4

**Review by sparshg — CHANGES_REQUESTED** [review]

File: manim/animation/movement.py (hunk 1 above)
Comment (behackl) [inline manim/animation/movement.py:57]: I like the suggestion of @sparshg, but dislike that the square is already deformed in the initial step. I have an alternate suggestion of a more classical example of a homotopy:

```suggestion
    .. manim:: HomotopyExample

        import math

        class HomotopyExample(Scene):
            def construct(self):
                square = Square()
                square.insert_n_curves(20)  # improve quality of homotopy target
                homotopy_map = lambda x, y, z, t: (
                    (1-t)*x + t*2*np.cos(math.atan2(y, x)),
                    (1-t)*y + t*2*np.sin(math.atan2(y, x)),
                    z,
                )
                self.play(Homotopy(homotopy_map, square))
```

Comment (behackl) [reply manim/animation/movement.py:57]: Output:

https://user-images.githubusercontent.com/11851593/201829134-db90993c-5945-41e9-a3f6-951ab7cf530a.mp4

What do you think?

**Comment by rli596** [comment]
Having a more interesting shape to apply the homotopy to is good, but I think the homotopy itself should be kept as mathematically simple as possible. This was my original reasoning behind choosing a dot too, but it might have been too simplistic.

But I like the square to circle deformation as a mathematically classical example of homotopy.

**Comment by rli596** [comment]
There's another use case which is very different to traditional homotopy: dynamic solutions. For example, a function of the form f(x,t) can be implemented using

homotopy = lambda x,y,z,t: np.array([x, f(x,t), 0])

applied to say, a straight line y = constant. Would it be worth including an example for this use case?

**Comment by behackl** [comment]
> There's another use case which is very different to traditional homotopy: dynamic solutions. For example, a function of the form f(x,t) can be implemented using
> 
> homotopy = lambda x,y,z,t: np.array([x, f(x,t), 0])
> 
> applied to say, a straight line y = constant. Would it be worth including an example for this use case?

Definitely, it would be nice to have a second small, perhaps easier to read example!

**Comment by JasonGrace2282** [comment]
Closing due to a lack of a response, feel free to reopen.

**Pull request closed without merging**